CHANGELOG.md

```diff
@@ -4,6 +4,18 @@ All notable changes to this project will be documented in this file.
 Each new release typically also includes the latest modulesync defaults.
 These should not affect the functionality of the module.
 
+## [v3.1.0](https://github.com/voxpupuli/puppet-ldapquery/tree/v3.1.0) (2023-11-15)
+
+[Full Changelog](https://github.com/voxpupuli/puppet-ldapquery/compare/v3.0.0...v3.1.0)
+
+**Implemented enhancements:**
+
+- Add Puppet 8 support [\#48](https://github.com/voxpupuli/puppet-ldapquery/pull/48) ([bastelfreak](https://github.com/bastelfreak))
+
+**Merged pull requests:**
+
+- Add replacement ldapquery::search function [\#47](https://github.com/voxpupuli/puppet-ldapquery/pull/47) ([alexjfisher](https://github.com/alexjfisher))
```

Comment:
I am wondering if the changes deserve more visibility than that (Merged pull requests are generally innocent)?

It seems that we add deprecation info in the doc but the code is not adding them and puppet will not complain about it.  Per se, a minor release at some point with the deprecation messages will be necessary before we roll a new major version that drop the legacy functions (or we roll a new major version with the deprecation in place and remove the deprecated functions in a 2nd major version?).

Reply:
I deliberately thought we could go another release or two before adding any stronger deprecation declared in the code itself.  Maybe I'll create a release summary ticket to add a bit more prominence though.